File: T3/codigos1920/proyectos/T3_PasoDatos/app/src/main/java/com/borja/t3_pasodatos/SecondActivity.java
package com.borja.t3_pasodatos;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

import com.borja.t3_pasodatos.utils.Persona;

public class SecondActivity extends AppCompatActivity {

    TextView textoRecuperar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);
        instancias();
        recuperarDatos();
    }

    private void recuperarDatos() {
        if (getIntent().getExtras() != null) {
            //String palabraRecuperada = (String) getIntent().getExtras().get(MainActivity.TAG_1);
            Persona pRecuperada = null;
            try{
                pRecuperada = (Persona) getIntent().getExtras().get(MainActivity.TAG_1);
            } catch (ClassCastException e){
                // FALLO
            }

            String saludo = String.format("Perfecto %s %s, tu teléfono es %d",
                    pRecuperada.getNombre(), pRecuperada.getApellido(), pRecuperada.getTelefono());
            textoRecuperar.setText(saludo);
        }
    }

    private void instancias() {
        textoRecuperar = findViewById(R.id.textoRecuperado);
    }
}

File: Kotlin/RECUPERACION/Inicio/app/src/main/java/com/example/inicio/SecondActivity.kt
package com.example.inicio

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.inicio.databinding.ActivitySecondBinding

class SecondActivity : AppCompatActivity() {

    private lateinit var binding: ActivitySecondBinding
    private lateinit var nombre: String;
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySecondBinding.inflate(layoutInflater)
        setContentView(binding.root)

        nombre = intent.getStringExtra("nombre") ?: "por defecto"
        binding.textoRecuperar.text = nombre;

    }
}